Mage.Sets/src/mage/cards/d/Deification.java

```diff
@@ -0,0 +1,148 @@
+
+    private SubType subType = null;
+
+    public FilterDeification() {
+        super("planeswalkers of the chosen type");
+    }
+
+    private FilterDeification(final FilterDeification filter) {
+        super(filter);
+        this.subType = filter.subType;
+    }
+
+    @Override
+    public FilterDeification copy() {
+        return new FilterDeification(this);
+    }
+
+    @Override
+    public boolean match(Permanent permanent, UUID playerId, Ability source, Game game) {
+        if (super.match(permanent, playerId, source, game)) {
+            if (subType == null) {
+                subType = ChoosePlaneswalkerTypeEffect.getChosenPlaneswalkerType(source.getSourceId(), game);
+                if (subType == null) {
+                    return false;
+                }
+            }
+            return permanent.hasSubtype(subType, game);
+        }
+        return false;
+    }
+
+}
+
+// Based on SerraTheBenevolentEmblemEffect
+class DeificationReplacementEffect extends ReplacementEffectImpl {
+
+    private static final Logger logger = Logger.getLogger(CardImpl.class);
+
+    DeificationReplacementEffect() {
+        super(Duration.Custom, Outcome.Benefit);
+        staticText = "As long as you control a creature, if damage dealt to a planeswalker you control of the chosen " +
+                "type would result in all loyalty counters on it being removed, instead all but one of those " +
+                "counters are removed.";
+    }
+
+    private DeificationReplacementEffect(final DeificationReplacementEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public DeificationReplacementEffect copy() {
+        return new DeificationReplacementEffect(this);
+    }
+
+    @Override
+    public boolean checksEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.REMOVE_COUNTERS;
+    }
+
+    @Override
+    public boolean applies(GameEvent event, Ability source, Game game) {
+
+        RemoveCountersEvent rEvent = (RemoveCountersEvent) event;
+
+        if (!source.isControlledBy(rEvent.getPlayerId())) {
+            return false;
+        }
+        Permanent planeswalker = game.getPermanentOrLKIBattlefield(rEvent.getTargetId());
+        if (planeswalker == null) {
+            return false;
+        }
+        if (!rEvent.counterRemovedDueToDamage()){
+            // not due to damage, prevention does not occur
+            return false;
+        }
+
+        int loyaltyCounters = planeswalker.getCounters(game).getCount(CounterType.LOYALTY);
+        if (planeswalker.hasSubtype(ChoosePlaneswalkerTypeEffect.getChosenPlaneswalkerType(source.getSourceId(), game), game)
+                && (loyaltyCounters - event.getAmount()) < 1
+                && game.getBattlefield().count(
+                StaticFilters.FILTER_CONTROLLED_CREATURE,
+                event.getPlayerId(), source, game) > 0
+        ) {
+            event.setAmount(loyaltyCounters - 1);
```

Comment:
this setAmount should be in `replaceEvent`, applies is for static checks only

Reply:
digging through the calls to `applies` and `replaceEvent`, im not 100% sure what the use case is for `applies` is--return true if this effect /would/ replace this event if you called `replaceEvent`?

Reply:
`applies` is for static quick checks as to whether the event should be replaced at all, if it returns false, then `replaceEvent` is never called

`replaceEvent` changes the event, returns false if event still exists (as it does here), true only if the event is fully consumed by the effect.

Reply:
I see, so it sounds like everything in this part of what is currently in `applies` should be moved to `replaceEvent`, yes? keeping the rest in `applies`

```java
int loyaltyCounters = planeswalker.getCounters(game).getCount(CounterType.LOYALTY);
if (planeswalker.hasSubtype(ChoosePlaneswalkerTypeEffect.getChosenPlaneswalkerType(source.getSourceId(), game), game)
        && (loyaltyCounters - event.getAmount()) < 1
        && game.getBattlefield().count(
        StaticFilters.FILTER_CONTROLLED_CREATURE,
        event.getPlayerId(), source, game) > 0
) {
    event.setAmount(loyaltyCounters - 1);
}
```

Reply:
No, the checks should be in applies. It's the `setAmount` that needs to be moved.